src/api/contract/unlock.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Call unlock method on wrapper contract
+ */
+module.exports = async (efx, token, amount) => {
+  const currency = efx.CURRENCIES[token]
+
+  // value we asking to unlock
+  const value = amount * (10 ** currency.decimals)
+
+  const {
+    signature,
+    unlockTill
+  } = await efx.sign.lock(currency.address)
+
+  const r = signature.r || 0
+  const s = signature.s || 0
+  const v = signature.v || 0
+
+  const args = [v, r, s, value, unlockTill]
```

Comment:
The signature required to unlock with the withdraw function should not be signed by the user's own key.

Instead it needs to be signed by Ethfinex. To do this, the user needs to request a signature from the releaseTokens endpoint

The signature will contain: keccak256(userAddress, wrapperAddress, signatureValidUntilBlock)

Reply:
@plutoegg do i need to send a "unlockUntil" to the http endpoint /releaseTokens ?

Reply:
I'm not exactly sure what i should use as  "signatureValidUntilBlock" on my test file